tests/tskey

```diff
@@ -30,37 +28,107 @@ else
     fi
 fi
 
-title PARA "Test Symmetric Encryption on the token (legacy init)"
-ossl 'enc -e -aes128
-             -K 0102030405060708090a0b0c0d0e0f10
-             -iv 100f0e0d0c0b0a090807060504030201
-             -in ${TMPPDIR}/aes128cbc.txt
-             -out ${TMPPDIR}/aes128cbc.txt.enc' "${PROP}"
-
-ossl 'enc -d -aes128
-             -K 0102030405060708090a0b0c0d0e0f10
-             -iv 100f0e0d0c0b0a090807060504030201
-             -in ${TMPPDIR}/aes128cbc.txt.enc
-             -out ${TMPPDIR}/aes128cbc.txt.dec' "${PROP}"
-
-diff "${TMPPDIR}/aes128cbc.txt" "${TMPPDIR}/aes128cbc.txt.dec"
-
-if [[ "${RUN_SKEYMGMT_TEST}" = "1" ]]; then
-title PARA "Test Symmetric Encryption on the token (skey init)"
-ossl 'enc -e -aes128 -skeymgmt AES
-             -skeyopt hexraw-bytes:0102030405060708090a0b0c0d0e0f10
-             -iv 100f0e0d0c0b0a090807060504030201
-             -in ${TMPPDIR}/aes128cbc.txt
-             -out ${TMPPDIR}/aes128cbc.txt.skey.enc' "${PROP}"
-
-ossl 'enc -d -aes128 -skeymgmt AES
-             -skeyopt hexraw-bytes:0102030405060708090a0b0c0d0e0f10
-             -iv 100f0e0d0c0b0a090807060504030201
-             -in ${TMPPDIR}/aes128cbc.txt.skey.enc
-             -out ${TMPPDIR}/aes128cbc.txt.skey.dec' "${PROP}"
-
-diff "${TMPPDIR}/aes128cbc.txt" "${TMPPDIR}/aes128cbc.txt.skey.dec"
-fi
+iv_needed() {
+    mode=$(echo "$1" |tr '[:upper:]' '[:lower:]')
+    res=1
+    if echo "$mode" |grep -qE "^(cbc|cfb|ofb|ctr)"; then
+        res=0
+    fi
+    return $res
+}
+
+#
+# Function to test a block cipher mode
+test_block_mode() {
+    local mode_name="$1" # mode, e.g., "CBC"
+    local cipher="$2" # ossl cipher without mode, e.g., "aes-128"
+    local input_len="$3" # plaintext length to test
+    local expected_outlen="$4" # expected output length
+    local key_mng="$5" # key management, "legacy" or "skey"
+
+    # generate plaintext of length $input_len
+    printf 'Plaintext%.0s' {1..100} |head -c "$input_len" >"${TMPPDIR}"/plaintext
+
+    mode_lower=$(echo "$mode_name" |tr '[:upper:]' '[:lower:]')
```

Comment:
I would put here a comment why this is done this way rather than the obvious bash shortcut to prevent people trying to simplify that.